src/algorithm/centroid.rs

```diff
@@ -0,0 +1,69 @@
+pub use types::{Coordinate, Point, LineString};
+
+pub trait Centroid {
+    fn centroid(&self) -> Point;
+}
+
+impl Centroid for LineString {
+    fn centroid(&self) -> Point {
+        let sum = self.0
+                      .iter()
+                      .fold(Point(Coordinate { x: 0., y: 0. }), |sum, &p| sum + p);
+        let size = self.0.len() as f64;
```

Comment:
hum I think the casting should be done only in the else (maybe directly in the  `/`: `x: sum.lng() / size as f64,` )